projects/ids-enterprise-ng/src/lib/toolbar-flex/soho-toolbar-flex.spec.ts

```diff
@@ -0,0 +1,214 @@
+  `;
+
+  model = {
+    searchValue: ''
+  };
+
+  searchfieldOptions = {
+    filterMode: 'contains',
+  };
+
+  onBeforeMoreMenuOpen = (response: AjaxBeforeOpenResponseFunction, options: any) => {
+    if (options.hasOwnProperty('contextElement')) {
+      if (options.contextElement[0].parentElement.parentElement.id === 'menu-button') {
+        response(this.MENU_BUTTON_RESPONSE_HTML);
+        return;
+      }
+      response(this.SUBMENU_RESPONSE_HTML);
+      return;
+    } else {
+      response(this.MENU_RESPONSE_HTML);
+      return;
+    }
+  }
+
+  onBeforeMenuButtonOpen = (response: AjaxBeforeOpenResponseFunction, options: any) => {
+    response(this.MENU_BUTTON_RESPONSE_HTML);
+    return;
+  }
+
+  onSelected(event) {
+    let data = '';
+    if (event.item.type === 'actionbutton' || event.item.type === 'menubutton') {
+      data = event.item.selectedAnchor[0].dataset.action;
+    } else {
+      data = event.item.element.dataset.action;
+    }
+    console.log(data);
+  }
+}
+
+describe('Soho Toolbar Flex Render', () => {
+  let toolbarflex: SohoToolbarFlexComponent;
+  let component: SohoToolbarFlexTestComponent;
+  let fixture: ComponentFixture<SohoToolbarFlexTestComponent>;
+  let de: DebugElement;
+  let el: HTMLElement;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [SohoToolbarFlexTestComponent],
+      imports: [FormsModule, SohoToolbarFlexModule, SohoMenuButtonModule]
+    });
+
+    fixture = TestBed.createComponent(SohoToolbarFlexTestComponent);
+    component = fixture.componentInstance;
+
+    toolbarflex = component.toolbarFlex;
+
+    de = fixture.debugElement;
+    el = de.query(By.css('soho-toolbar-flex')).nativeElement;
+  });
+
+  it('Check HTML content', () => {
+    fixture.detectChanges();
+
+    expect(el.nodeName).toEqual('SOHO-TOOLBAR-FLEX');
+    expect(el.children.length).toEqual(4);
+
+    const firstSection = el.children[0];
+    expect(firstSection.nodeName).toEqual('SOHO-TOOLBAR-FLEX-SECTION');
+    expect(firstSection.classList).toContain('toolbar-section');
+    expect(firstSection.classList).toContain('title');
+
+    const secondSection = el.children[1];
+    expect(secondSection.nodeName).toEqual('SOHO-TOOLBAR-FLEX-SECTION');
+    expect(secondSection.classList).toContain('toolbar-section');
+    expect(secondSection.classList).toContain('buttonset');
+
+    const thirdSection = el.children[2];
+    expect(thirdSection.nodeName).toEqual('SOHO-TOOLBAR-FLEX-SECTION');
+    expect(thirdSection.classList).toContain('toolbar-section');
+    expect(thirdSection.classList).toContain('search');
+
+    const fourthSection = el.children[3];
+    expect(fourthSection.nodeName).toEqual('SOHO-TOOLBAR-FLEX-MORE-BUTTON');
+    expect(fourthSection.classList).toContain('toolbar-section');
+    expect(fourthSection.classList).toContain('more');
+  });
+
+  describe('Check events', () => {
```

Comment:
would also like to see check inputs test and verification of update() being called properly and also a test for public functions. 

See soho-pie.spec.ts as a pretty good example. 

https://github.com/infor-design/enterprise-ng/blob/76b1a43a4e7c4d56b09f0a15640342a334d09941/projects/ids-enterprise-ng/src/lib/pie/soho-pie.spec.ts#L76

https://github.com/infor-design/enterprise-ng/blob/76b1a43a4e7c4d56b09f0a15640342a334d09941/projects/ids-enterprise-ng/src/lib/pie/soho-pie.spec.ts#L156